src/SprykerEco/Yves/Braintree/Theme/default/components/molecules/braintree-payment-form/braintree-payment-form.ts

```diff
@@ -0,0 +1,169 @@
+        const nonceInputSelector = <HTMLInputElement>document.querySelector(`input[name='${this.nonceInputName}']`);
+        nonceInputSelector.value = nonce;
+    }
+
+    protected clearTokenValue(): void {
+        const nonceInputSelector = <HTMLInputElement>document.querySelector(`input[name='${this.nonceInputName}']`);
+
+        nonceInputSelector.value = '';
+    }
+
+    protected errorHandler(error: braintreeErrorConfig): void {
+        const paymentMethod = this.currentPaymentMethodValue;
+
+        this.removeErrorMessage();
+
+        if (paymentMethod === this.paymentMethodName) {
+            this.addErrorMessage(error.message);
+            return;
+        }
+
+        this.submitForm();
+    }
+
+    protected paymentMethodHandler(response: braintreeConfig): void {
+        const paymentMethod = this.currentPaymentMethodValue;
+        const isWrongMethodSelected = (paymentMethod === this.paymentMethodName && response.type !== this.paymentMethodTypeName);
+
+        this.removeErrorMessage();
+
+        if (isWrongMethodSelected) {
+            this.errorHandler({
+                message: this.braintreeErrorMessage
+            });
+            return;
+        }
+
+        this.submitForm(response.nonce);
+    }
+
+    protected loadBraintree(): void {
+        this.braintreeSetupSettings = {
+            onPaymentMethodReceived: this.paymentMethodHandler.bind(this),
+            onError: this.errorHandler.bind(this)
+        };
+    }
+
+    protected setupBraintree(): void {
+        this.loadBraintree();
+        braintree.setup(this.braintreeClientToken, this.integrationType, this.braintreeSetupSettings);
+    }
+
+    setCurrentPaymentMethod(): void {
+        this.paymentMethods.forEach((method: HTMLInputElement) => {
+            this.setCurrentPaymentMethodValue(method);
+        });
+    }
+
+    setCurrentPaymentMethodValue(method: HTMLInputElement): void {
+        if (method.checked) {
+            this.currentPaymentMethodValue = method.value;
+        }
+    }
+
+    addErrorMessage(message: string = ''): void {
+        const errorContainer = <HTMLElement>this.querySelector(`.${this.jsName}__error`);
+        this.showErrorMessage(errorContainer);
+
+        errorContainer.innerHTML = message;
+    }
+
+    showErrorMessage(errorContainer: HTMLElement): void {
+        errorContainer.classList.remove('is-hidden');
+    }
+
+    removeErrorMessage(): void {
+        const errorContainer = <HTMLElement>this.querySelector(`.${this.jsName}__error`);
+        this.hideErrorMessage(errorContainer);
+        errorContainer.innerHTML = '';
+    }
+
+    hideErrorMessage(errorContainer: HTMLElement): void {
+        errorContainer.classList.add('is-hidden');
+    }
+
+    get braintreeClientToken(): string {
+        return this.getAttribute('data-braintree-client-token');
+    }
+
+    get braintreeErrorMessage(): string {
+        return this.getAttribute('data-error-message');
+    }
+}
```

Comment:
please add comments for all getters, public methods and public variables

Reply:
fixed